### PR TITLE
Add WAF policy management support

### LINKED ISSUE #1257: Add WAF policy management support
### Current Terraform Version
1.26.0

### Use-cases

- Create policies.
- Support to modify the name, level and the mode of detection of an existing policy.
- Support to modify the configuration items of the existing policy.
- Support to modify the action of the existing policy.

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

- Add WAF policy management support.
	- Support the creation and management of policies.
	
	- Support the use of existing policies to create domain.
	
	  
	
- Add test cases to domain management: 

  - create a domain using an existing policy.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
resolves #1257

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

- Create a domain with an existing policy.

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafDomainV1_policy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafDomainV1_policy -timeout 360m -parallel 4
=== RUN   TestAccWafDomainV1_policy
=== PAUSE TestAccWafDomainV1_policy
=== CONT  TestAccWafDomainV1_policy
--- PASS: TestAccWafDomainV1_policy (29.11s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       29.194s
```

- Create policy and change policy.

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafPolicyV1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafPolicyV1 -timeout 360m -parallel 4
=== RUN   TestAccWafPolicyV1_basic
--- PASS: TestAccWafPolicyV1_basic (24.81s)
=== RUN   TestAccWafPolicyV1_update
--- PASS: TestAccWafPolicyV1_update (34.54s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       59.422s

```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
+package waf
+
+import (
+	"fmt"
+	"log"
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: use fmtp and logp packages, please

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+func resourceWafPolicyV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	wafClient, err := config.WafV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud WAF client: %s", err)
+	}
+
+	createOpts := policies.CreateOpts{
+		Name: d.Get("name").(string),
+	}
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_policy.go:144]: seems we do nothing about *level* and *protection_mode* parameters.

Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go:144]: added

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go (hunk 1 above)
Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: has been changed

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+	} else {
+		d.Set("protection_mode", protectionMode)
+		d.Set("level", level)
+	}
+
+	return err
+}
+
+func checkAndUpdateDefaultVal(policy *policies.Policy, protectionMode string, level int,
+	wafClient *golangsdk.ServiceClient) error {
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: suggest to set the golangsdk.ServiceClient as the first parameter

File: huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_policy_test.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+					resource.TestCheckResourceAttr(resourceName, "protection_mode", "log"),
+					resource.TestCheckResourceAttr(resourceName, "level", "2"),
+					resource.TestCheckResourceAttr(resourceName, "full_detection", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccWafPolicyV1_update(t *testing.T) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_policy_test.go]: suggest to discard `TestAccWafPolicyV1_basic` and reneme this function to `TestAccWafPolicyV1_basic`

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go (hunk 3 above)
Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: done

File: huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_policy_test.go (hunk 4 above)
Comment (chengxiangdong) [reply huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_policy_test.go]: done

**Pull request merged**